vyper/venom/passes/sccp/eval.py

```diff
@@ -0,0 +1,129 @@
+import operator
+from typing import Callable
+
+from vyper.utils import SizeLimits, evm_div, evm_mod
+from vyper.venom.basicblock import IROperand
+
+
+def _unsigned_to_signed(value: int) -> int:
+    if value <= SizeLimits.MAX_INT256:
+        return value
+    else:
+        return value - SizeLimits.CEILING_UINT256
+
+
+def _signed_to_unsigned(value: int) -> int:
+    if value >= 0:
+        return value
+    else:
+        return value + SizeLimits.CEILING_UINT256
+
+
+def _wrap_int_binop(operation):
+    def wrapper(ops: list[IROperand]) -> int:
+        first = _unsigned_to_signed(ops[1].value)
+        second = _unsigned_to_signed(ops[0].value)
+        return _signed_to_unsigned(int(operation(first, second)))
+
+    return wrapper
+
+
+def _wrap_uint_binop(operation):
```

Comment:
how about `_unsigned_binop`

Reply:
or even just `_binop` (and we know by convention it is unsigned) as most of our binops are unsigned